### ISSUE TITLE
Retrain model and Background mislabeled

### ISSUE THREAD

**Issue by rlu25** [Deep-MI/FastSurfer#688]
## Description

I used the OASIS as training data. The label I use the FastSurfer prediction. The problem is, when I used the data retrain the model, the testing mislabeled the background. 
## Steps to Reproduce

1. Predict MRI.nii.gz (from OASIS) using uploaded model, get label.mgz
2. Use ori.mgz and label.mgz as training data
3. The train prediction and val prediction mislabeled the background

...

## Expected Behavior

The train and val prediction should same as the label but background be labeled

## Log Files /Screenshots

I attached the log and .yaml for axial
## Environment
 - Local install
 - dev-FastSurferCNN
 - Linux
 - GPU
 

### Execution

1. Run run_prediction.py to get label
2. Run generate_dhf5.py to get training dataset
3. Run run_model.py for training

## Additional Context

axial.yaml
{
  "MODEL": {
    "MODEL_NAME": "FastSurferVINN",
    "NUM_CLASSES": 79,
    "LOSS_FUNC": "combined",
    "NUM_FILTERS": 71,
    "NUM_FILTERS_INTERPOL": 32,
    "NUM_BLOCKS": 5,
    "NUM_CHANNELS": 7,
    "KERNEL_H": 3,
    "KERNEL_W": 3,
    "KERNEL_C": 1,
    "STRIDE_CONV": 1,
    "STRIDE_POOL": 2,
    "POOL": 2,
    "HEIGHT": 256,
    "WIDTH": 256,
    "BASE_RES": 1.0,
    "INTERPOLATION_MODE": "bilinear",
    "CROP_POSITION": "top_left",
    "OUT_TENSOR_WIDTH": 320,
    "OUT_TENSOR_HEIGHT": 320
  },
  "TRAIN": {
    "BATCH_SIZE": 8,
    "LOG_INTERVAL": 50,
    "RESUME": false,
    "RESUME_EXPR_NUM": "Default",
    "NUM_EPOCHS": 30,
    "NUM_STEPS": 10,
    "FINE_TUNE": false,
    "CHECKPOINT_PERIOD": 2,
    "NUM_WORKERS": 8,
    "EARLY_STOPPING": true,
    "EARLY_STOPPING_MODE": "min",
    "EARLY_STOPPING_PATIENCE": 10,
    "EARLY_STOPPING_WAIT": 10,
    "EARLY_STOPPING_DELTA": 1e-05
  },
  "TEST": {
    "BATCH_SIZE": 16
  },
  "DATA": {
    "PATH_HDF5_TRAIN": "/opt/localdata/data/usr-envs/ruiying/Code/NeurIPS/FastSurfer/FastSurfer-train/FastSurferCNN/train_data/training_set_axial.hdf5",
    "PATH_HDF5_VAL": "/opt/localdata/data/usr-envs/ruiying/Code/NeurIPS/FastSurfer/FastSurfer-train/FastSurferCNN/train_data/training_set_axial.hdf5",
    "PLANE": "axial",
    "CLASS_OPTIONS": [
      "aseg",
      "aparc"
    ],
    "SIZES": [
      256,
      311,
      320
    ],
    "PADDED_SIZE": 320,
    "AUG": [
      "None",
      "Gaussian"
    ]
  },
  "DATA_LOADER": {
    "NUM_WORKERS": 3,
    "PIN_MEMORY": true
  },
  "OPTIMIZER": {
    "BASE_LR": 0.01,
    "LR_SCHEDULER": "cosineWarmRestarts",
    "GAMMA": 0.3,
    "STEP_SIZE": 5,
    "ETA_MIN": 0.0001,
    "T_ZERO": 10,
    "T_MULT": 2,
    "FACTOR": 0.1,
    "PATIENCE": 5,
    "THRESH": 0.0001,
    "COOLDOWN": 0,
    "MOMENTUM": 0.9,
    "DAMPENING": 0.0,
    "NESTEROV": true,
    "WEIGHT_DECAY": 0.0001,
    "OPTIMIZING_METHOD": "adamW"
  },
  "NUM_GPUS": 1,
  "LOG_DIR": "/opt/localdata/data/usr-envs/ruiying/Code/NeurIPS/FastSurfer/FastSurfer-train/FastSurferCNN/experiments/axial/FastSurferVINN",
  "EXPR_NUM": "FastSurferVINN",
  "RNG_SEED": 1,
  "SUMMARY_PATH": "/opt/localdata/data/usr-envs/ruiying/Code/NeurIPS/FastSurfer/FastSurfer-train/FastSurferCNN/experiments/axial/FastSurferVINN/summary/FastSurferVINN",
  "CONFIG_LOG_PATH": "/opt/localdata/data/usr-envs/ruiying/Code/NeurIPS/FastSurfer/FastSurfer-train/FastSurferCNN/experiments/axial/FastSurferVINN/config/FastSurferVINN"
}
[FastSurferVINN.log](https://github.com/user-attachments/files/19660697/FastSurferVINN.log)

<img width="642" alt="Image" src="https://github.com/user-attachments/assets/b3f2e561-22d5-447f-a97c-a29727dcea19" />

**Comment by dkuegler** [comment]
These results look very weird. for one, there seem to be different intensity levels of teal. Depending on your dataset, Epoch 30 is not really that long. FastSurferVINN is usually trained for 100+ epochs.

Can you maybe answer a couple of questions?
Do you have made any modifications to the training script?
What label are the teal and purple regions?

Is this just the prediction of one network or multiple networks?
Please highlight differences of the network configuration to the defaults.

**Comment by rlu25** [comment]
> These results look very weird. for one, there seem to be different intensity levels of teal. Depending on your dataset, Epoch 30 is not really that long. FastSurferVINN is usually trained for 100+ epochs.
> 
> Can you maybe answer a couple of questions? Do you have made any modifications to the training script? What label are the teal and purple regions?
> 
> Is this just the prediction of one network or multiple networks? Please highlight differences of the network configuration to the defaults.

Hi,
I didn't change the training script or defaults. I only use the label which I got from FastSurfur as training data.
The teal labeled 54 and purple labeled 5.
I predict using multiple networks. All networks predict the background randomly.

**Comment by dkuegler** [comment]
That sounds a bit like the weights in the dataset are not configured properly.
This is practically impossible for us to help you with in detail. But maybe have a look at weights item [here](https://github.com/Deep-MI/FastSurfer/blob/4dddadf0b2ec2ab32d1b2e7ae0b664827836e9f1/FastSurferCNN/train.py#L128).

**Comment by rlu25** [comment]
> That sounds a bit like the weights in the dataset are not configured properly. This is practically impossible for us to help you with in detail. But maybe have a look at weights item [here](https://github.com/Deep-MI/FastSurfer/blob/4dddadf0b2ec2ab32d1b2e7ae0b664827836e9f1/FastSurferCNN/train.py#L128).

You're night, the weights cause the background makes no contribution for the loss function. I tried to ignore the weights, result is good.

**Comment by dkuegler** [comment]
I dont know how you have configured you max_weights ín the dataset [generation](https://github.com/Deep-MI/FastSurfer/blob/4dddadf0b2ec2ab32d1b2e7ae0b664827836e9f1/FastSurferCNN/generate_hdf5.py#L305), but you need to make sure, that the background weight exists (even if it is small). Otherwise the label for the background will essentiallly be random.

**Comment by rlu25** [comment]
> I dont know how you have configured you max_weights ín the dataset [generation](https://github.com/Deep-MI/FastSurfer/blob/4dddadf0b2ec2ab32d1b2e7ae0b664827836e9f1/FastSurferCNN/generate_hdf5.py#L305), but you need to make sure, that the background weight exists (even if it is small). Otherwise the label for the background will essentiallly be random.

Thanks for the explanation!
